backend/controller/webAuthn.go

```diff
@@ -266,15 +268,44 @@ func AuthorizeResponse(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Could not finish log in", http.StatusInternalServerError)
 		return
 	}
-
-	err = SaveDBCredential(w, credential)
-	if err != nil {
+	//err = SaveDBCredential(w, credential)
+	//if err != nil {
+	//	http.Error(w, "Could not finish log in", http.StatusInternalServerError)
+	//	return
+	//}
+
+	user := models.User{}
+	creds := models.Credential{Id: credential.ID}
+	res := db.DB.First(&creds)
+	if res.Error != nil {
+		http.Error(w, "Could retrieve credentials ", http.StatusInternalServerError)
 		return
 	}
-
-	errJson := json.NewEncoder(w).Encode("Login Success")
+	db.DB.Model(&creds).Association("User").Find(&user)
+	roles := []models.Role{}
+	db.DB.Model(&user).Association("Roles").Find(&roles)
+	doors := []models.Door{}
+	db.DB.Distinct("id").Model(&roles).Association("Doors").Find(&doors)
+	hasAccess := false
+	for _, curDoor := range doors {
+		if curDoor.Id == dId {
+			hasAccess = true
+		}
```

Comment:
Let's break if we find the id

Reply:
Actually, can we just use a built-in search function: https://stackoverflow.com/questions/38654383/how-to-search-for-an-element-in-a-golang-slice
https://pkg.go.dev/slices#IndexFunc